src/endpoints/mex/mex.token.charts.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from "@nestjs/common";
+import { GraphQlService } from "src/common/graphql/graphql.service";
+import { OriginLogger } from "@multiversx/sdk-nestjs-common";
+import { gql } from 'graphql-request';
+import { MexTokenChart } from "./entities/mex.token.chart";
+import { MexTokenService } from "./mex.token.service";
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { CacheInfo } from "src/utils/cache.info";
+
+@Injectable()
+export class MexTokenChartsService {
+  private readonly logger = new OriginLogger(MexTokenChartsService.name);
+
+  constructor(
+    private readonly graphQlService: GraphQlService,
+    private readonly mexTokenService: MexTokenService,
+    private readonly cachingService: CacheService,
+  ) { }
+
+  async getTokenPricesHourResolution(tokenIdentifier: string): Promise<MexTokenChart[]> {
+    return await this.cachingService.getOrSet(
+      CacheInfo.TokenHourChar(tokenIdentifier).key,
+      async () => await this.getTokenPricesHourResolutionRaw(tokenIdentifier),
+      CacheInfo.TokenHourChar(tokenIdentifier).ttl,
+    );
+  }
+
+  async getTokenPricesHourResolutionRaw(tokenIdentifier: string): Promise<MexTokenChart[]> {
+    const tokenExists = await this.checkTokenExists(tokenIdentifier);
+
+    if (!tokenExists) {
+      this.logger.error(`Token ${tokenIdentifier} does not exist or has no trading pair`);
```

Comment:
does it worth logging this for each user bad request?

Reply:
fixed it, I used undefined if the given token is not part of mex tokens. Also, if undefined, from mex controller I throw NotFoundException